dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java

```diff
@@ -167,6 +167,13 @@ public Map<String, Object> insertSchedule(User loginUser,
         executorService.checkProcessDefinitionValid(projectCode, processDefinition, processDefineCode,
                 processDefinition.getVersion());
 
+        Schedule scheduleExists =
+                scheduleMapper.queryByProcessDefinitionCode(processDefineCode);
+        if (scheduleExists != null) {
+            throw new ServiceException(Status.SCHEDULE_ALREADY_EXISTS, processDefineCode,
```

Comment:
Could we put a message in the result to tell the reason to the user, like below:
```
if (scheduleExists != null) {
  log.error("xxx");
  putMsg(result, Status.XXX);
  return result;
}
```

Reply:
It seems already okay.